eng/native/functions.cmake

```diff
@@ -423,7 +423,7 @@ function(strip_symbols targetName outputFilename)
         VERBATIM
         COMMAND ${DSYMUTIL} ${DSYMUTIL_OPTS} ${strip_source_file}
         COMMAND ${strip_command}
-        COMMENT "Stripping symbols from ${strip_source_file} into file ${strip_destination_file}"
```

Comment:
Does this apply to other cases like preprocess_file? Also, I don't find many such cases. Do you know what prevents the generator expansion?

Reply:
These two are the only places where comment has generator expressions.

With `COMMENT`, cmake inserts the string verbatim when generating the project files at configure-time, i.e., if the string has generator expressions (`$<...>`), they are not evaluated. In order to obtain the resolved value of `TARGET_FILE` (or any generator expression), it requires delegating the string to shell utilities (`echo`, `printf` etc.) via `COMMAND`.

The absolute paths from `$<TARGET_FILE:...>` are quite long here; deep nesting under `/prefix/runtime/artifacts/obj`, hence the filename without path is chosen for logs.

To obtain the filenames without path, shell's `basename` is used instead of cmake's `$<TARGET_NAME:...>` because the destination path is calculated by another function (`get_symbol_file_name`) and `strip_symbols` is operating on variables with generator expressions. Therefore, `TARGET_NAME` will require refactoring both functions. OTOH, `basename` is a simpler solution and equally portable (it's a POSIX-y utility and code is under Unix condition).